src/artifact_service/service.rs

```diff
@@ -898,10 +894,10 @@ mod tests {
             .await
             .unwrap_err();
 
-        assert_eq!(
-            error,
-            BuildError::InitializationFailed("No authorized nodes found".to_owned())
-        );
+        match error {
+            BuildError::InitializationFailed(msg) => assert_eq!(msg, "No authorized nodes found"),
+            _ => panic!("Invalid BuildError: {}", error),
```

Comment:
It seems to be this as this error does not implement `Display` but uses `derive(Debug)` annotation.
```suggestion
            _ => panic!("Invalid BuildError: {:?}", error),
```

Reply:
Actually it implements `Display`. Please refer to https://docs.rs/thiserror/latest/thiserror/index.html